includes/admin/class-wc-admin-addons.php

```diff
@@ -476,18 +479,46 @@ public static function output_featured_sections( $sections ) {
 		}
 	}
 
+	/**
+	 * Returns in-app-purchase URL params.
+	 */
+	public static function get_in_app_purchase_url_params() {
+		$back_admin_path = isset( $_SERVER['REQUEST_URI'] ) ? esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
```

Comment:
We should try and avoid using $_SERVER vars, perhaps use `add_query_arg( array(), $wp->request )` instead?

Reply:
@kloon Looked at this, and I couldn't get anything to work. `$wp->request` is mostly empty on the relevant admin pages – maybe it's not intended for wp-admin (admin prefix is not included, would have to add it back with `admin_url`)? In any case, it doesn't seem to contain the "file" url component, e.g. `edit.php`. See `print_r` dumps of `request` below.

I also tried using `WP_Screen`, and this looked promising, but I couldn't find a way to convert the screen object to a url or a url path. Any ideas? 

### orders page (empty state)
<img width="412" alt="orders" src="https://user-images.githubusercontent.com/4167300/61189260-06972f80-a6df-11e9-84dc-a843d66cc393.png">

### product edit page
<img width="480" alt="product-edit" src="https://user-images.githubusercontent.com/4167300/61189262-0e56d400-a6df-11e9-8df8-9de0791a7278.png">

### WooCommerce > Extensions screen
<img width="344" alt="Screen Shot 2019-07-15 at 8 58 20 AM" src="https://user-images.githubusercontent.com/4167300/61189266-1878d280-a6df-11e9-88b8-8bd2c1200001.png">

Reply:
Aha! Now using `add_query_arg` (thanks for the tip on slack). Fixed in dab7cb328ad7e5d9122404502daf35f19ed2a989